### DISCUSSION TITLE
Issue with creating cross-repository pull requests in same organization

### DISCUSSION THREAD

**Discussion by subesokun** [community/community#40034] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hi,

I want to use the REST API to create cross-repository pull requests but I always get the error that my `head` is invalid.

I've the following setup:

Repo: `myorg/myrepo`
Fork: `myorg/myrepo-fork`
Branch: `myorg/myrepo-fork:test`
Now I want to merge `myorg/myrepo-fork:test` into `myorg/myrepo:main` via

```
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  /repos/myorg/myrepo/pulls \
  -f title='Amazing new feature' \
 -f body='Please pull these awesome changes in!' \
 -f head='myorg/myrepo-fork:test' \
 -f base='main'

{
  "message": "Validation Failed",
  "errors": [
    {
      "resource": "PullRequest",
      "field": "head",
      "code": "invalid"
    }
  ],
  "documentation_url": "https://docs.github.com/rest/reference/pulls#create-a-pull-request"
}
```

I've tried also `myrepo-fork:test` as head but no luck. What am I doing wrong?

Thanks!

**Comment by LangLangBart** [comment]
did u try it with `test` ?

```zsh
gh api \
  --method POST \
  -H "Accept: application/vnd.github+json" \
  /repos/myorg/myrepo/pulls \
  -f title='Amazing new feature' \
 -f body='Please pull these awesome changes in!' \
 -f head='test' \
 -f base='main'
```

**Reply by subesokun** [reply]
Same error :(

**Reply by LangLangBart** [reply]
- check with `GH_DEBUG=api` in front of your query.
- your branch is on GitHub or locally ?

**Reply by subesokun** [reply]
My branch is on GitHub and it contains mergable changes. In fact, creating the PR via the Web UI works but I want to do it via the REST API. Creating a PR from a branch within the same repo worked via the GH CLI, so permissions should be fine.

**Reply by LangLangBart** [reply]
Seems not possible, tried `REST API` and `GraphQL API` both unsuccessful.
Then I stumbled across this issue and it confirms that it is not possible.

[cli/cli - Issue #6462 - gh and REST API unable to work with same-org forked repos](https://github.com/cli/cli/issues/6462) (19/Oct/22)

**Reply by subesokun** [reply]
Thanks for looking that up! Hope that this API will be soon available 🤞

**Comment by subesokun** [comment, marked as answer]
There is now an API available for that https://github.com/cli/cli/issues/6462#issuecomment-1397267993
